app/src/androidTest/java/ch/epfl/sdp/blindly/chat/ChatAdapterTest.kt

```diff
@@ -0,0 +1,47 @@
+package ch.epfl.sdp.blindly.chat
+
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import junit.framework.Assert.assertEquals
+import junit.framework.Assert.assertNotSame
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class ChatAdapterTest {
+
+    @Test
+    fun recyclerViewShowsCorrectCount() {
+        val m1 = Message("mes1", "2938473dsgfd298")
+        val m2 = Message("mes2", "293s73dshfd298")
+        val list: ArrayList<Message> = arrayListOf<Message>()
+        list.addAll(listOf(m1, m2))
+        val abc: RecyclerView.Adapter<ChatAdapter.ViewHolder> = ChatAdapter(list)
+        abc.notifyDataSetChanged()
+        assertEquals(2, abc.itemCount)
+    }
+
+    @Test
+    fun itemTypeIsCorrect() {
+        val m1 = Message("mes1", "2938473dsgfd298")
+        val m2 = Message("mes2", "293s73dshfd298")
+        val list: ArrayList<Message> = arrayListOf<Message>()
+        list.addAll(listOf(m1, m2))
+        val abc: RecyclerView.Adapter<ChatAdapter.ViewHolder> = ChatAdapter(list)
+        assertEquals(1, abc.getItemViewType(1))
+    }
+
+
+    @Test
+    fun messageObjectIsCorrect() {
```

Comment:
The name of the test is not really representative of what it does in my opinion. Maybe move the timestamp test of line 41 in another test function and consider only m2 in this test (as m1 is used only in the timestamp test).